repl/src/main/scala/com/cloudera/livy/repl/SparkContextInitializer.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to Cloudera, Inc. under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Cloudera, Inc. licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cloudera.livy.repl
+
+import org.apache.spark.{SparkConf, SparkContext}
+
+import com.cloudera.livy.Logging
+
+/**
+ * A mixin trait for Spark entry point creation. This trait exists two different code path
+ * separately for Spark1 and Spark2, depends on whether SparkSession exists or not.
+ */
+trait SparkContextInitializer extends Logging {
+  self: SparkInterpreter =>
+
+  def createSparkContext(conf: SparkConf): Unit = {
+    if (isSparkSessionPresent()) {
+      spark2CreateContext(conf)
+    } else {
+      spark1CreateContext(conf)
+    }
+  }
+
+  private def spark1CreateContext(conf: SparkConf): Unit = {
+    val sparkClz = classOf[SparkContext]
+    sparkContext = sparkClz.getMethod("getOrCreate", classOf[SparkConf]).invoke(null, conf)
```

Comment:
Do you need reflection for `SparkContext` methods? Aren't they binary compatible between 1.6 and 2.0? The signature in the source code is the same.